core/engine_test.go

```diff
@@ -968,3 +968,45 @@ func TestMinIterationDuration(t *testing.T) {
 	// But we expect the custom counter to be added to 4 times
 	assert.Equal(t, 4.0, getMetricSum(collector, "testcounter"))
 }
+
+func TestMinIterationDurationInSetupTeardownStage(t *testing.T) {
+	t.Parallel()
+
+	runner, err := js.New(
+		&loader.SourceData{URL: &url.URL{Path: "/script.js"}, Data: []byte(`
+		import { sleep } from "k6";
+
+		export function setup() {
+			sleep(1);
+		}
+
+		export let options = {
+			minIterationDuration: "2s",
+			duration: "2s",
+			setupTimeout: "2s",
+		};
+
+		export default function () {
+		};`)},
```

Comment:
We should also test `teardown()`, no? It likely hits the same code path, but we should confirm the fix for both since it's mentioned in the issue.

Reply:
Fixed